### DISCUSSION TITLE
hidewearablesoftype does not hide hair when hidelimb is true

### DISCUSSION THREAD

**Discussion by 4-S-H** [FakeFishGames/Barotrauma#14688] [Code questions and modding support]
Might be related to https://github.com/FakeFishGames/Barotrauma/issues/11562;

Code of the wearable (the sprite is transparent):
```
<Wearable limbtype="Head" slots="Any,Headset" msg="ItemMsgPickUpSelect" displaycontainedstatus="true">
  <sprite name="mod_Head" texture="%ModDir%/images.png" limb="Head" hidelimb="true" inheritlimbdepth="true" ignorelimbscale="true" hidewearablesoftype="Moustache,Beard,Hair,FaceAttachment,Husk,Herpes" sourcerect="0,0,256,256" origin="0.5,0.5" />
</Wearable>
```

Appearance of the head without the wearable:
![image](https://github.com/user-attachments/assets/3cd14942-a48c-4e2d-a054-f8c18a7704e5)
with the wearable and `hidelimb=false`:
![image](https://github.com/user-attachments/assets/c267c940-ce4e-4927-a94b-d88b7a675561)
with the wearable and `hidelimb=true`:
![image](https://github.com/user-attachments/assets/5d828a5e-ed26-4229-af19-d72e8c18a4e4)

The code here might be responsible: when the if check on line 891 succeeds and the one on line 893 fails, the original hair sprite is drawn even though it should be hidden.
https://github.com/FakeFishGames/Barotrauma/blob/0e8fb6569d2810e2f8ad5fb17b4bba546cc5739a/Barotrauma/BarotraumaClient/ClientSource/Characters/Limb.cs#L886-L908

**Comment by 4-S-H** [comment]
Maybe this is intended but I dont know how to work around this :(

**Comment by Regalis11** [comment]
This has now been addressed in our private repo, fix coming up in the next unstable build and the upcoming release. Sorry for the inconvenience!
